src/tests/trainers.test.js

```diff
@@ -1,3 +1,274 @@
+      dni: 43576575,
+    };
+    const response = await request(app).post('/api/trainers').send(reqBody);
+    expect(response.status).toBe(400);
+    expect(response.body.message).toBeDefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('Should return The admin with this Email already exists', async () => {
+    const reqBody = {
+      email: 'delfi@gmail.com',
+    };
+    const response = await request(app).post('/api/trainers').send(reqBody);
+    expect(response.status).toBe(400);
+    expect(response.body.message).toBeDefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('Should return invalid name format', async () => {
+    const reqBody = {
+      firstName: '123Gino',
+    };
+    const response = await request(app).post('/api/admins').send(reqBody);
+    expect(response.body.data).toBeUndefined();
+    expect(response.status).toBe(400);
+    expect(response.error).toBeTruthy();
+  });
+
+  test('Should return invalid phone number fortmat', async () => {
+    const reqBody = {
+      phone: '8383',
+    };
+    const response = await request(app).post('/api/admins').send(reqBody);
+    expect(response.body.data).toBeUndefined();
+    expect(response.status).toBe(400);
+    expect(response.error).toBeTruthy();
+  });
+
+  test('Should return invalid Password format', async () => {
+    const reqBody = {
+      password: 'strongepassword123',
+    };
+    const response = await request(app).post('/api/admins').send(reqBody);
+    expect(response.body.data).toBeUndefined();
+    expect(response.status).toBe(400);
+    expect(response.error).toBeTruthy();
+  });
+
+  test('Should return invalid salary format', async () => {
+    const reqBody = {
+      salary: 'salary',
+    };
+    const response = await request(app).post('/api/admins').send(reqBody);
+    expect(response.body.data).toBeUndefined();
+    expect(response.status).toBe(400);
+    expect(response.error).toBeTruthy();
+  });
+
+  test('Should return an status 400', async () => {
+    const reqBody = {};
+    const response = await request(app).post('/api/trainers').send(reqBody);
+    expect(response.status).toBe(400);
+    expect(response.body.message).toBeDefined();
+    expect(response.body.error).toBeTruthy();
+  });
+});
+
+describe('DELETE /api/trainers', () => {
+  test('Should create a trainer with a status 200', async () => {
+    const id = '646642acfac4c6a035b35000';
+    const response = await request(app).delete(`/api/trainers/${id}`);
+    expect(response.status).toBe(200);
+    expect(response.body.message).toBeDefined();
+    expect(response.body.error).toBeFalsy();
+  });
+
+  test('Should return ID not found with a status 404', async () => {
+    const id = '646642acfac4c6a035b35002';
+    const response = await request(app).delete(`/api/trainers/${id}`);
+    expect(response.status).toBe(404);
+    expect(response.body.message).toBeDefined();
+    expect(response.body.error).toBeTruthy();
+  });
```

Comment:
You could try deleting the same id on the previous test to make sure it's actually deleted and not found. Like: 
test('Should return ID not found with a status 404', async () => {
    const id = '646642acfac4c6a035b35000';
    const response = await request(app).delete(`/api/trainers/${id}`);
    expect(response.status).toBe(404);
    expect(response.body.message).toBeDefined();
    expect(response.body.error).toBeTruthy();
  });